src/lib/y2remote/remote.rb

```diff
@@ -0,0 +1,233 @@
+
+      @modes.delete(:vnc) if mode == :manager
+      @modes.delete(:manager) if mode == :vnc
+
+      @modes << mode
+    end
+
+    # Whether the vnc manager mode is enabled or not
+    #
+    # @return [Boolean] true it the :manager mode is enabled
+    def with_manager?
+      modes.include?(:manager)
+    end
+
+    # Read the current status of vnc and the enabled modes
+    #
+    # @return [Boolean] true
+    def read
+      if xdm_enabled? && display_manager_remote_access?
+        @modes = Y2Remote::Modes.running_modes
+      end
+
+      true
+    end
+
+    # Update the SCR according to vnc settings
+    #
+    # @return [Boolean] true on success
+    def write
+      steps = [_("Configure display manager")]
+      steps << _("Restart the services") if Yast::Mode.normal
+
+      caption = _("Saving Remote Administration Configuration")
+
+      Yast::Progress.New(caption, " ", steps.size, steps, [], "")
+      Yast::Progress.NextStage
+      Yast::Progress.Title(_("Configuring display manager..."))
+      return false unless configure_display_manager
+
+      if Yast::Mode.normal
+        Yast::Progress.NextStage
+        Yast::Progress.Title(_("Restarting the service..."))
+        restart_services
+        Yast::Progress.NextStage
+      end
+
+      true
+    end
+
+    # Reset the proposal configuration enabling vnc if it was enabled by
+    # linuxrc
+    def reset!
+      @proposed = false
+
+      propose!
+    end
+
+    # It propose the vnc configuration if it has not been proposed yet
+    #
+    # @return [Boolean]
+    def propose!
+      return false if proposed?
+
+      Yast::Linuxrc.vnc ? enable! : disable!
+
+      log.info("Remote Administration was proposed as: #{modes.inspect}")
+
+      @proposed = true
+    end
+
+    # Updates the VNC and xdm configuration
+    #
+    # Called from #write. Ensures that required packages are installed,
+    # enables vnc services and xdm and writes the configuration files,
+    # reporting any error in the process.
+    #
+    # @return [Boolean] true if success, false otherwise
+    def configure_display_manager
+      if enabled?
+        # Install required packages
+        if !Yast::Package.InstallAll(required_packages)
+          log.error "Installing of required packages failed"
+          return false
+        end
+
+        Y2Remote::Modes.all.each { |m| modes.include?(m) ? m.enable! : m.disable! }
+      end
+
+      # Set DISPLAYMANAGER_REMOTE_ACCESS in sysconfig/displaymanager
+      Yast::SCR.Write(
+        Yast.path(".sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS"),
+        enabled? ? "yes" : "no"
+      )
+      Yast::SCR.Write(
+        Yast.path(".sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE"),
+        enabled? ? "yes" : "no"
+      )
+      Yast::SCR.Write(Yast.path(".sysconfig.displaymanager"), nil)
+
+      true
+    end
+
+    def restart_display_manager
+      if Yast::Service.active?(XDM_SERVICE_NAME)
+        Yast::Report.Error(
```

Comment:
np: As you report the very same error a few lines below, what about moving it to a separate method and make this line more readable?

Reply:
Completely agree, have not touched it yet at all, I was even thinking in add a class for DisplayManager which handle this and other logic.